src/tech_assistant_agent/tools.py

```diff
@@ -98,78 +95,47 @@ async def confirm_memorized_fields(
     context: TurnContext,
 ) -> str:
     print("Confirming memorized fields", fields_to_confirm)
-    if not fields_to_confirm.fields_to_confirm:
+    if not fields_to_confirm.fields:
         print("No fields to confirm")
         return "No fields to confirm"
-    flattened_memory_ids = [
-        memory_id
-        for user_detail in fields_to_confirm.fields_to_confirm
-        for memory_id in user_detail.memory_ids
-    ]
-    if not flattened_memory_ids:
-        return "No memories to confirm"
-    memories = await memory_module.get_memories(memory_ids=flattened_memory_ids)
-    if not memories:
-        return "No memories to confirm"
-    # group memories by field name
-    user_details_with_memories: List[tuple[UserDetail, Memory | None]] = []
-    for user_detail in fields_to_confirm.fields_to_confirm:
-        memories_for_user_detail = [
-            memory for memory in memories if memory.id in user_detail.memory_ids
-        ]
-        # just take the first one into account for citation (for now)
-        user_details_with_memories.append(
-            (
-                user_detail,
-                memories_for_user_detail[0] if memories_for_user_detail else None,
-            )
-        )
 
-    cited_memories: List[Memory] = [
-        memory for _, memory in user_details_with_memories if memory is not None
-    ]
-    # Get all message IDs from memory attributions
-    message_ids = []
-    for memory in cited_memories:
-        if memory.message_attributions:
-            message_ids.extend(memory.message_attributions)
-
-    messages = await memory_module.get_messages(message_ids)
-    # Create a lookup dict for messages by ID
-    messages_by_id = {msg.id: msg for msg in messages}
+    # Get memories and attributed messages
+    cited_fields = []
+    for user_detail in fields_to_confirm.fields:
+        field_name = user_detail.field_name
+        field_value = user_detail.field_value
+        memories_with_citations = None
+        if user_detail.memory_ids:
+            memories_with_citations = await memory_module.get_memories_with_citations(
+                memory_ids=user_detail.memory_ids
+            )
+        cited_fields.append((field_name, field_value, memories_with_citations))
```

Comment:
NIT: thought about flattening this first? Then calling "get_memories_with_citations"?